components/shell/fish/patches/path_max.patch

```diff
@@ -0,0 +1,13 @@
+diff --git a/src/wutil.cpp b/src/wutil.cpp
+index 5b69c47f..9c4bedd2 100644
+--- a/src/wutil.cpp
++++ b/src/wutil.cpp
+@@ -139,7 +139,7 @@ bool wreaddir_for_dirs(DIR *dir, wcstring *out_name) {
+ const wcstring wgetcwd() {
+     wcstring retval;
+ 
+-    char *res = getcwd(NULL, 0);
++    char *res = getcwd(NULL, PATH_MAX);
```

Comment:
I've tested with PATH_MAX + 1 and still no error.  So if you think that PATH_MAX should be + 1 then i will update the patch